### DISCUSSION TITLE
Can I use entity.columnId?

### DISCUSSION THREAD

**Discussion by SDAdham** [mikro-orm/mikro-orm#1210] [Q&A]
**Is your feature request related to a problem? Please describe.**
Is it possible to use `entity.columnId` instead of `entity.table.id`?
I tried to do this but it didn't work with me, can you please help?

**Describe the solution you'd like**

**Describe alternatives you've considered**
For me to just set an id, I have to do this:
```
entity.someColumn = 'xyz';
entity.table = new Table();
entity.table.id = 2; // to link the row to another row of another table.
```

**Additional context**
Add any other context about the feature request here.

**Comment by B4nan** [comment]
Lol, this is not how you should use issue template 😭 You left 80% of the hints there and literally hidden your question inside 🤷 

You can use `mapToPk`, but it will have consequences (propagation of changes won't work). 

> For me to just set an id, I have to do this:

Nope, actually that is not even the correct way. If the entity with ID 2 exists, this is wrong way to go, as it would result in insert query. You should use `entity.table = em.getReference(Table, 2);` instead.

**Reply by SDAdham** [reply]
thanks b4nan. Sorry for not filling out the 80%, your answer is right on spot for my question. I think I'll be able to continue from here.

**Reply by SDAdham** [reply]
please close this issue.

**Reply by SDAdham** [reply]
@B4nan if it's ok, can you please clarify what do you mean by propagation of changes won't work?

**Reply by SDAdham** [reply]
Below is how i'm hoping to use the `userId` as an example.
```
  userId: number;

  @ManyToOne(
    () => User,
    { nullable: true} ,
  )
  user: User;
```
Based on the screenshot below from the doc here https://mikro-orm.io/docs/defining-entities/#mapping-directly-to-primary-keys, there is no example of using both class and column name directly
![image](https://user-images.githubusercontent.com/11182824/102076261-96ed1d00-3e5b-11eb-96b9-ff3118220912.png)

**Reply by B4nan** [reply]
https://mikro-orm.io/docs/propagation/

The issue with `mapToPk` is mainly if you use autoincrement PK, as then it is not possible to do:

```ts
const author = new Author();
const book = new Book();
book.author = author;
em.persist(book);
await em.flush();
```

Normally this would flush both book and the author, if you would want to map to PK instead on `Book.author`, you would have to flush the author first and assign its PK:

```ts
const author = new Author();
await em.persist(book).flush(); // flushing author first to obtain its PK
const book = new Book();
book.author = author.id; // assigning the PK
await em.persist(book).flush();
```

**Reply by SDAdham** [reply]
Ahh ok, that makes sense. But how can i define the `mapToPk` with such entity below:
```
  userId: number;

  @ManyToOne(
    () => User,
    { nullable: true }
  )
  user: User;
```

I am guessing it would be like this?
```
@ManyToOne(
    () => User,
    { nullable: true, mapToPk: true }
  )
  userId: number;

  @ManyToOne(
    () => User,
    { nullable: true, mapToPk: true }
  )
  user: User;
```

**Reply by B4nan** [reply]
> there is no example of using both class and column name directly

Because that is not supported, only if you make one of the properties virtual (via `persist: false` in the decorator).

Correct way is to use `em.getReference()` to assign to relations, `mapToPk` is not a silver bullet, and is not designed to be used as "another way to assign to the relation".

**Reply by SDAdham** [reply]
Thanks B4nan, I'll use `em.getReference()` but any chance we can add this suggestions please?

**Reply by yf-hk** [reply]
@SDAdham Currently it's doable. 
See https://github.com/mikro-orm/mikro-orm/issues/1469#issuecomment-783188128
